astropy/modeling/rotations.py

```diff
@@ -187,9 +187,6 @@ def evaluate(cls, x, y, angle):
         single coordinate pair.
         """
 
-        if x.shape != y.shape:
-            raise ValueError("Expected input arrays to have the same shape")
```

Comment:
It would fail without the check, yes. But this gives an opportunity to raise a nicer error message.  Another option would be to change it to a try/except, which would be fine.  But I'm working on fixing the root issue.